### PR TITLE
svcacct: support jose.OpaqueSigner and push errors to token generator creation

### PR DESCRIPTION
**Pull request by mikedanese**
```release-note
NONE
```

### CODE REVIEW

**Comment by spiffxp** [comment]
@mikedanese is this still needed?

File: pkg/serviceaccount/jwt.go [hunk 1]
```diff
@@ -54,21 +54,9 @@ type TokenGenerator interface {
 // JWTTokenGenerator returns a TokenGenerator that generates signed JWT tokens, using the given privateKey.
```
Comment (awly) [inline pkg/serviceaccount/jwt.go]: `s/privateKey/skey/g` or just rename the argument back to `privateKey`

File: pkg/serviceaccount/jwt.go [hunk 2]
```diff
@@ -80,25 +68,40 @@ func (j *jwtTokenGenerator) GenerateToken(claims *jwt.Claims, privateClaims inte
 		case elliptic.P521():
 			alg = jose.ES512
 		default:
-			return "", fmt.Errorf("unknown private key curve, must be 256, 384, or 521")
+			return nil, fmt.Errorf("unknown private key curve, must be 256, 384, or 521")
 		}
 	default:
-		return "", fmt.Errorf("unknown private key type %T, must be *rsa.PrivateKey or *ecdsa.PrivateKey", j.privateKey)
+		if osig, ok := skey.(jose.OpaqueSigner); ok {
```
Comment (awly) [inline pkg/serviceaccount/jwt.go]: make this a regular case in the type switch

File: pkg/serviceaccount/jwt.go [hunk 3]
```diff
@@ -80,25 +68,40 @@ func (j *jwtTokenGenerator) GenerateToken(claims *jwt.Claims, privateClaims inte
...
+			alg = jose.SignatureAlgorithm(osig.Public().Algorithm)
+			break
+		}
+		return nil, fmt.Errorf("unknown private key type %T, must be *rsa.PrivateKey or *ecdsa.PrivateKey", skey)
```
Comment (awly) [inline pkg/serviceaccount/jwt.go]: ... or jose.OpaqueSigner

**Review by awly — COMMENTED** [review]
LGTM, just a few stylistic comments

**Comment by mikedanese** [comment]
/assign @cblecker 

for dep approval

**Comment by mikedanese** [comment]
@awly Fixed.

File: pkg/serviceaccount/jwt.go [hunk 4]
```diff
@@ -54,51 +54,52 @@ type TokenGenerator interface {
...
 			alg = jose.ES512
 		default:
-			return "", fmt.Errorf("unknown private key curve, must be 256, 384, or 521")
+			return nil, fmt.Errorf("unknown private key curve, must be 256, 384, or 521")
 		}
+	case jose.OpaqueSigner:
+		alg = jose.SignatureAlgorithm(pk.Public().Algorithm)
 	default:
-		return "", fmt.Errorf("unknown private key type %T, must be *rsa.PrivateKey or *ecdsa.PrivateKey", j.privateKey)
+		return nil, fmt.Errorf("unknown private key type %T, must be *rsa.PrivateKey or *ecdsa.PrivateKey", privateKey)
```
Comment (awly) [inline pkg/serviceaccount/jwt.go]: mention jose.OpequeSigner in the message

**Comment by cblecker** [comment]
Bump of existing dep. No licence changes.
/approve

**Comment by mikedanese** [comment]
@awly fixed.

**Pull request merged**